app/models/manageiq/providers/amazon/inventory/parser/cloud_manager.rb

```diff
@@ -206,7 +215,7 @@ def instance_hardware(persister_instance, instance, flavor)
       :cpu_total_cores      => flavor[:cpus],
       :memory_mb            => flavor[:memory] / 1.megabyte,
       :disk_capacity        => flavor[:ephemeral_disk_size],
-      :guest_os             => persister.hardwares.lazy_find(instance['image_id'], :key => :guest_os),
+      :guest_os             => persister.hardwares.lazy_find(persister.vms.lazy_find(instance['image_id']), :key => :guest_os),
```

Comment:
so, I am still keeping the old way that we do not need to define a key, if the manager_ref is size 1

so
```persister.hardwares.lazy_find(persister.vms.lazy_find(instance['image_id']))```
is equivalent to:
```persister.hardwares.lazy_find(:vm_or_temaplte => persister.vms.lazy_find(:ems_ref => instance['image_id']))```

Or would you prefer just the hash syntax?